packages/mui-material/src/TablePagination/TablePagination.d.ts

```diff
@@ -32,6 +32,37 @@ export interface TablePaginationTypeMap<P, D extends React.ElementType> {
        * Override or extend the styles applied to the component.
        */
       classes?: Partial<TablePaginationClasses>;
+      /**
+       * The components used for First, Last, Next & Previous item type
+       *
+       * This prop is an alias for the `components` prop, which will be deprecated in the future.
+       *
+       * @default {
+       *   First: FirstPageIcon,
+       *   Last: LastPageIcon,
+       *   Next: KeyboardArrowRight,
+       *   Previous: KeyboardArrowLeft,
+       * }
+       */
+      slots?: {
+        First?: React.ElementType;
+        Last?: React.ElementType;
+        Next?: React.ElementType;
+        Previous?: React.ElementType;
+      };
+      /**
+       * The props used for each slot inside.
+       *
+       * This prop is an alias for the `componentsProps` prop, which will be deprecated in the future.
+       *
+       * @default {}
+       */
+      slotProps?: {
+        first?: React.HTMLProps<HTMLButtonElement>;
```

Comment:
These are not buttons but icons. The type of slots fields should reflect this.

Besides, these slotProps should be in line with their MUI Base counterparts: https://github.com/mui/material-ui/blob/9f7a2bfac55faed1b1c6c7b4b6d1b1659a076bab/packages/mui-base/src/TablePaginationUnstyled/TablePaginationUnstyled.types.ts#LL44-L48C7

Reply:
> These are not buttons but icons. The type of slots fields should reflect this.

`SvgIconProps` should suffice the requirement, right?

> Besides, these slotProps should be in line with their MUI Base counterparts: https://github.com/mui/material-ui/blob/9f7a2bfac55faed1b1c6c7b4b6d1b1659a076bab/packages/mui-base/src/TablePaginationUnstyled/TablePaginationUnstyled.types.ts#LL44-L48C7

Sorry, but I did not understand, what exactly is expected here.

Reply:
> SvgIconProps should suffice the requirement, right?

Yep, I think this should work.

> Sorry, but I did not understand, what exactly is expected here.

Instead of having `first`, `last`, etc. as fields of the `slotProps` object, create the `actions` field, as in TablePaginationUnstyled.

Reply:
@michaldudak To match the MUI Base implementation, we need to decide whether we want to expose the `TablePaginationActions` component from Material UI to support the `actions` slot in table pagination. Currently, it is an internal component and is not exported from `TablePagination`, unlike `TablePaginationActionsUnstyled`, which is exported from `TablePaginationUnstyled`. There we can use the custom action slot and override  each icon using the actions unstyled component.

Otherwise, having the slots per icon on TablePagination is fine, I believe.

Reply:
@michaldudak, what are your thoughts? This PR has been pending for a long time.

Reply:
Hey! Just catching up to this conversation. Let me know if I'm missing something from the context.

I think what @michaldudak proposed was adding the `actions` slot field in the `TablePagination` component. If we only added that we would be able to achieve the [same behavior as this example](https://mui.com/material-ui/react-table/#custom-pagination-actions) but using `slots.actions` instead of `ActionsComponent`.

I think the solution we're looking for is adding the `actions` slot field to `TablePagination`, but also adding what this PR is adding to `TablePaginationActions`: the `first`, `last`, `next`, and `previous` slots. So the user could do:

```tsx
<TablePagination
  //...
  slotProps={
    actions: {
      slots: {
        first: IconComponent,
      },
    },
  }
/>
```

The mental model here would be providing the `slots` prop to the `TablePaginationActions` component using the `TablePagination` `slotProps` prop. No need to export `TablePaginationActions`.

With this we could also deprecate `TablePagination`'s `ActionsCommponent` prop as the `slots.actions` slot would be the standard way to achieve this.

Does that make sense? That's consistent with Base UI, with [`PaginationItem`](https://mui.com/material-ui/react-pagination/#custom-icons), and adds the desired functionality of it making it easier to provide the icons without having to rebuild the entire actions component.

@pratikkarad a thing that we need to change is the `TablePaginationActions` `slots` casing, for example, `First` -> `first`.

Thanks for the contribution by the way, sorry for the long time this is taking. I hope we can close it soon 😊

Reply:
Sorry, this got lost among notifications.

My aim is to reduce the number of breaking changes we'll have to do when switching the internal implementation to Base UI. If we can match the structure from Base UI's TablePagination (including making TablePaginationActions public), we won't have to change it drastically with v6.